### PR TITLE
chore: sort the denpendencies

### PR DESCRIPTION
**Pull request by ben1009**
### All Submissions:

* [x] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you formatted your code locally using `cargo +nightly fmt --all` command prior to submission?
3. [x] Have you checked your code using `cargo clippy --all --all-features` command?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?	
- run [cargo sort -g](https://crates.io/crates/cargo-sort) to sort  the  dependencies in Cargo.toml alphabetically.dependencies
* [ ] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -34,78 +34,97 @@ tokio-tracing = ["tokio/tracing"]
...
-clap = { version = "4.4.18", features = ["derive"] }
+itertools = "0.12"
+log = "0.4"
+memory = { path = "lib/common/memory" }
+parking_lot = { version = "0.12.1", features = [
+    "deadlock_detection",
+], optional = true }
+prometheus = { version = "0.13.3", default-features = false }
+prost = "0.11.9"
+# Consensus related crates
```
Comment (timvisee) [inline Cargo.toml:77]: Leaving this comment here does not make sense now.

File: Cargo.toml [hunk 2]
```diff
@@ -34,78 +34,97 @@ tokio-tracing = ["tokio/tracing"]
...
-cancel = { path = "lib/common/cancel" }
-memory = { path = "lib/common/memory" }
-segment = { path = "lib/segment" }
-collection = { path = "lib/collection" }
-storage = { path = "lib/storage" }
-api = { path = "lib/api" }
-actix-multipart = "0.6.1"
-constant_time_eq = "0.3.0"
 
 # Profiling
```
Comment (timvisee) [inline Cargo.toml:117]: Same, this is an useless comment now.

**Review by generall — CHANGES_REQUESTED** [review]
The benefit of sorting dependencies is kinda questionable. In current version we could have some kind of grouping.
In my opinion sorting only makes sense if we want to search faster, which doesn't look like a problem in case of cargo files.

I would only keep the "remove unused" part

**Comment by ben1009** [comment]
> The benefit of sorting dependencies is kinda questionable. In current version we could have some kind of grouping. In my opinion sorting only makes sense if we want to search faster, which doesn't look like a problem in case of cargo files.
> 
> I would only keep the "remove unused" part

get your point

- revert to run `cargo sort -g` to keep the group if any
- for sort besides search faster, another benefit is [Large projects tend to like for things to stay sorted to cut down the occurrence of merge conflicts between two developers appending to the end of the same unsorted list in parallel changes. ](https://github.com/dtolnay/request-for-implementation/issues/29)
- for the "remove unused" part, already done in https://github.com/qdrant/qdrant/pull/3435, as you can see, no cargo.lock change in this pr

**Comment by timvisee** [comment]
> for sort besides search faster, another benefit is [Large projects tend to like for things to stay sorted to cut down the occurrence of merge conflicts between two developers appending to the end of the same unsorted list in parallel changes. ](https://github.com/dtolnay/request-for-implementation/issues/29)

Yes, I do like this part, but we can still do that with grouping.

Removing unused and sorting groups would be a good middle ground for me :+1:

**Pull request closed without merging**